packages/core/src/util/properties-util.ts

```diff
@@ -17,8 +17,8 @@ export const nonEnumerable = (target: any, propertyKey: string) => {
  * @returns A boolean denoting whether an entity is a navigation property or not.
  */
 export function isNavigationProperty<EntityT extends EntityBase>(
-  key: keyof EntityT,
-  entityConstructor: Constructable<EntityT>
+  key: keyof EntityT | string,
+  entityConstructor: Constructable<EntityT> | any
```

Comment:
is `any` a left over?

Reply:
No it is on purpose but I can probably improve the signature here